packages/tools/nodeEditor/src/components/preview/previewManager.ts

```diff
@@ -159,6 +160,44 @@ export class PreviewManager {
 
         this._lightParent = new TransformNode("LightParent", this._scene);
 
+        this._globalState.filesInput = new FilesInput(
+            this._engine,
+            this._scene,
+            (_, scene) => {
+                this._meshes.push(...scene.meshes);
+                this._prepareScene();
+            },
+            null,
+            null,
+            null,
+            null,
+            null,
+            () => {
+                this._globalState.previewType = PreviewType.Box;
+                this._globalState.listOfCustomPreviewFiles = [];
+                this._scene.meshes.forEach((m) => m.dispose());
+                this._globalState.onRefreshPreviewMeshControlComponentRequiredObservable.notifyObservers();
+                this._refreshPreviewMesh(true);
+            },
+            true
+        );
+        const canvas = this._engine.getRenderingCanvas();
+        if (canvas) {
+            const onDrag = (evt: DragEvent) => {
+                evt.stopPropagation();
+                evt.preventDefault();
+            };
+            canvas.addEventListener("dragenter", onDrag, false);
+            canvas.addEventListener("dragover", onDrag, false);
+
+            const onDrop = (evt: DragEvent) => {
+                evt.stopPropagation();
+                evt.preventDefault();
+                this._globalState.onDropEventReceivedObservable.notifyObservers(evt);
+            };
+            canvas.addEventListener("drop", onDrop, false);

```

Comment:
this entire part can be the responsibility of the filesinput module. It listens to the events and triggers the loadFiles. Unless there is a reason not to use it?

Reply:
I wanted both the drag and drop listener and the already existing file selection to take the same path (on PreviewMeshControlComponent.useCustomMesh), so that's why I didn't use the monitorForDragAndDrop, as it always calls loadFiles.